src/manifolds/SymmetricPositiveDefiniteGeneralizedBuresWasserstein.jl

```diff
@@ -0,0 +1,108 @@
+
+Compute the distance with respect to the [`BuresWassersteinMetric`](@ref) on [`SymmetricPositiveDefinite`](@ref) matrices, i.e.
+
+```math
+    d(p.q) = \bigl( \operatorname{tr}(M^{-1}p) + \operatorname{tr}(M^{-1}q) + 2\operatorname{tr}(p^{\frac{1}{2}M^{-1}qM^{-1}p^{\frac{1}[2}) \bigr)^\frac{1}[2},
+```
+
+where the last trace can be simplified (by rotating the matrix products in the trace) to ``\operatorname{tr}(M^{-1}pM^{-1}q)``.
+"""
+function distance(
+    M::MetricManifold{ℝ,SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric},
+    p,
+    q,
+)
+    luM = lu(M.metric.M)
+    return sqrt(tr(luM \ p) + tr(luM \ q) + 2 * tr((lum \ q) * (lum \ p)))
+end
+
+@doc raw"""
+    exp(::MatricManifold{ℝ,SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric}, p, X)
+
+Compute the exponential map on [`SymmetricPositiveDefinite`](@ref) with respect to
+the [`GeneralizedBuresWassersteinMetric`](@ref) given by
+
+```math
+    \exp_p(X) = p+X+\mathcal ML_{p,M}(X)pML_{p,M}(X)
+```
+
+where ``q=L_{M,p}(X)`` denotes the generalized lyaponov operator, i.e. it solves ``pqM + Mqp = X``.
+"""
+exp(::MetricManifold{ℝ,SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric}, p, X)
+
+function exp!(
+    M::MetricManifold{ℝ,SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric},
+    q,
+    p,
+    X,
+)
+    q .= lyapc(p, M.metric.M, -X) #lyap solves qpM + Mpq - X =0
+    q .= p .+ X .+ M.metricM * q * p * q * M.metric.M
+    return Y
+end
+
+@doc raw"""
+    inner(::MetricManifold{ℝ,SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric}, p, X, Y)
+
+Compute the inner product [`SymmetricPositiveDefinite`](@ref) with respect to
+the [`GeneralizedBuresWassersteinMetric`](@ref) given by
+
+```math
+    ⟨X,Y⟩_{\mathrm{GBW}} = \frac{1}{2}\operatorname{tr}( \mathcal L_{p,M}(X)Y)
+```
+
+where ``q=L_{M,p}(X)`` denotes the generalized lyaponov operator, i.e. it solves ``pqM + Mqp = X``.
+"""
+function inner(
+    M::MetricManifold{ℝ,SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric},
+    p,
+    X,
+    Y,
+)
+    return 1 / 2 * tr(lyapc(p, M.metric.M, -X) * Y)
+end
+
+@doc raw"""
+    loq(::MatricManifold{SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric}, p, q)
+
+Compute the logarithmic map on [`SymmetricPositiveDefinite`](@ref) with respect to
+the [`BuresWassersteinMetric`](@ref) given by
+
+```math
+    \loq_p(q) = M(M^{-1}pM^{-1}q)^{\frac{1}{2}} + (qM^{-1}pM^{-1})^{\frac{1}{2}}M^{-1} - 2 p
+```
+
+where ``q=L_p(X)`` denotes the lyaponov operator, i.e. it solves ``pq + qp = X``.
+"""
+log(::MetricManifold{ℝ,SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric}, p, q)
+
+function loq!(
+    M::MetricManifold{ℝ,SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric},
+    X,
+    p,
+    q,
+)
+    m = M.metric.M
+    lum = l(m)
+    X .= Symmetric(m * sqrt((lum \ p / lum) * q) + sqrt(q * (lum \ p / lum)) * m) - 2 * p
```

Comment:
```suggestion
    lum_p_lum = lum \ p / lum
    X .= Symmetric(m * sqrt(lum_p_lum * q) + sqrt(q * lum_p_lum) * m) - 2 * p
```
We don't have to compute that twice.